File: Backend/Microservices/Inventory/Eefa.Inventory.Domain/Aggregates/WarehouseAggregate/Warehouse.cs
using Eefa.Common;
using Eefa.Common.Domain;

namespace Eefa.Inventory.Domain
{
    using Eefa.Common.Data;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Collections.Generic;

    [Table("Warehouses", Schema = "inventory")]
    public partial class Warehouse : DomainBaseEntity, IAggregateRoot, IHierarchical
    {
        public int? ParentId { get; set; }
        /// <description>
        /// کد سطح
        ///</description>

        public string? LevelCode { get; set; }
        /// <description>
        /// عنوان
        ///</description>

        public string Title { get; set; } = default!;
        /// <description>
        /// فعال
        ///</description>

        public bool IsActive { get; set; } = default!;
        /// <description>
        /// کد گروه کالا
        ///</description>

        public int? CommodityCategoryId { get; set; } = default!;
        public int? AccountHeadId { get; set; } = default!;
        public int? AccountRererenceGroupId { get; set; } = default!;
        public int? AccountReferenceId { get; set; } = default!;
        public int? Sort { get; set; } = default!;
        public bool Countable { get; set; } = default!;

        /// <description>
        /// نقش صاحب سند
        ///</description>

        //public virtual CommodityCategory CommodityCategory { get; set; } = default!;
        //public virtual Warehouse? Parent { get; set; }
        //public virtual ICollection<Warehouse> InverseParent { get; set; } = default!;
        //public virtual ICollection<Stock> Stocks { get; set; } = default!;
        //public virtual ICollection<Receipt> Receipt { get; set; } = default!;
        //public virtual ICollection<WarehouseLayout> WarehouseLayouts { get; set; } = default!;
        public List<AuditMapRule> Audit()
        {
            return new List<AuditMapRule>();
        }
    }


}

File: Frontend/angular-app/src/app/modules/inventory/entities/warehouse.ts
import { CommodityCategory } from "../../commodity/entities/commodity-category";
import { ReceiptAllStatusModel } from "./receipt-all-status";

export class Warehouse {
  public  id!: number;
  public parentId!: number ;
  public levelCode!: string ;
  public title!: string;
  public isActive!: boolean;
  public commodityCategoryId!: number;
  public accessPermission!: string;
  public accessPermissionTitle!: string;
  public accountRererenceGroupId!: number;
  public accountReferenceId !: number;
  public accountHeadId !: number;
  public sort !: number;
  public countable!:boolean;
  public commodityCategories: CommodityCategory[] = [];
  public receiptAllStatus: ReceiptAllStatusModel[] = [];

}
